### PR TITLE
CBO metric

### PR DESCRIPTION
**Pull request by adriflorence**
CBO Metric calculates the **sum** of references from a class to other classes **and** the number of those classes that reference the class in question. 

For now, this also includes Exception types and Libraries.

### CODE REVIEW

**Review by moonbrew — CHANGES_REQUESTED** [review]
I suggested a few changes. See Mattermost.

File: src/com/metricsassignment/metrics/CBOMetric.java [hunk 1]
```diff
@@ -1,12 +1,61 @@
...
+
+        // every class that references this class
+        Integer referencedInOtherClasses = 0;
+        for( Set<String> refs : cboMap.values()){
+            if(refs.contains(classname)){
+                referencedInOtherClasses++;
+            }
+        }
+
+        return referenceToOtherClass + referencedInOtherClasses;
```
Comment (moonbrew) [inline src/com/metricsassignment/metrics/CBOMetric.java]: This would contain duplicates if a class referred to another class and that class referred back.

**Review by moonbrew — CHANGES_REQUESTED** [review]
Looks good. Just finish the final changes mentioned in mattermost.

**Review by moonbrew — APPROVED** [review]

File: src/com/metricsassignment/metrics/CBOMetric.java (hunk 1 above)
Comment (adriflorence) [reply src/com/metricsassignment/metrics/CBOMetric.java]: True, I put them into a set, so it should only consider them once even if it's a two-way reference.

**Comment by moonbrew** [comment]
I made the changes that I had suggested and then merged it. I have no other suggestions to make.

**Pull request merged**